src/lib/bits/pin-input/components/pin-input.svelte

```diff
@@ -0,0 +1,70 @@
+<script lang="ts">
+	import { melt } from "@melt-ui/svelte";
+	import { setCtx, getAttrs } from "../ctx.js";
+	import type { Props } from "../types.js";
+	import { derived } from "svelte/store";
+
+	type $$Props = Props;
+
+	export let placeholder: $$Props["placeholder"] = undefined;
+	export let value: $$Props["value"] = undefined;
+	export let name: $$Props["name"] = undefined;
+	export let disabled: $$Props["disabled"] = undefined;
+	export let type: $$Props["type"] = "text";
+	export let onValueChange: $$Props["onValueChange"] = undefined;
+	export let id: $$Props["id"] = undefined;
+	export let asChild: $$Props["asChild"] = false;
+
+	const {
+		elements: { root },
+		states: { value: localValue },
+		updateOption,
+		ids
+	} = setCtx({
+		placeholder,
+		defaultValue: value,
+		name,
+		disabled,
+		type,
+		onValueChange: ({ next }) => {
+			if (value !== next) {
+				onValueChange?.(next);
+				value = next;
+			}
+			return next;
+		}
+	});
+
+	$: value !== undefined && localValue.set(value);
+
+	const attrs = getAttrs("root");
+
+	$: updateOption("placeholder", placeholder);
+	$: updateOption("name", name);
+	$: updateOption("disabled", disabled);
+	$: updateOption("type", type);
+
+	$: builder = $root;
+	$: Object.assign(builder, attrs);
+
+	const idValues = derived([ids.root], ([$menubarId]) => ({
```

Comment:
tried to follow other examples but not sure what I am doing and why it's needed

Reply:
In Melt, each Id is its own store, and we want the `ids` slot prop object to always have the latest IDs. So we're deriving a new store from all the individual ID stores so as the individual ids update, so does this!